packages/react-vapor/src/components/pagination/PaginationPagesNumber.tsx

```diff
@@ -0,0 +1,156 @@
+
+    return {
+        currentPage: item ? item.pageNb : 0,
+    };
+};
+
+const mapDispatchToProps = (
+    dispatch: (action: IReduxAction<IReduxActionsPayload>) => void,
+    ownProps: INavigationPaginationOwnProps
+): INavigationPaginationDispatchProps => ({
+    onRender: () => dispatch(addPagination(ownProps.id)),
+    onDestroy: () => dispatch(removePagination(ownProps.id)),
+    onPageClick: (pageNb: number) => dispatch(changePage(ownProps.id, pageNb)),
+});
+
+export interface INavigationPaginationProps
+    extends IPaginationPagesNumberOwnProps,
+        ReturnType<typeof mapDispatchToProps>,
+        ReturnType<typeof mapStateToProps> {}
+
+export const NUMBER_OF_PAGES_SHOWING: number = 7;
+export const PREVIOUS_LABEL: string = 'Previous';
+export const NEXT_LABEL: string = 'Next';
+
+class PaginationPagesNumberDisconnected extends React.Component<INavigationPaginationProps, any> {
+    private handlePageClick = (pageNb: number) => {
+        if (pageNb >= 0 && this.props.currentPage !== pageNb) {
+            this.props.onPageClick?.(pageNb);
+        }
+    };
+
+    componentDidUpdate() {
+        if (this.props.currentPage > this.props.totalPages - 1) {
+            this.handlePageClick(this.props.totalPages - 1);
+        }
+    }
+
+    componentDidMount() {
+        this.props.onRender?.();
+    }
+
+    componentWillUnmount() {
+        this.props.onDestroy?.();
+    }
+
+    render() {
+        const currentPage: number = this.props.currentPage || 0;
+
+        const showXPages: number = Math.abs((this.props.numberOfPagesToShow || NUMBER_OF_PAGES_SHOWING) - 1);
+        const previousLabel: string = this.props.previousLabel || PREVIOUS_LABEL;
+        const nextLabel: string = this.props.nextLabel || NEXT_LABEL;
+        let start: number = 0;
+        let end: number = showXPages;
+        const lastPage: number = this.props.totalPages - 1;
+        const previousClasses: string = classNames('flat-select-option mod-link', {
+            disabled: currentPage === 0 || this.props.disabled,
+            selectable: currentPage !== 0,
+            hidden: this.props.hidden,
+        });
+        const nextClasses: string = classNames('flat-select-option mod-link', {
+            disabled: currentPage === lastPage || this.props.disabled,
+            selectable: currentPage !== lastPage,
+            hidden: this.props.hidden,
+        });
+
+        const pageSelects: JSX.Element[] = [];
+
+        if (!this.props.hidden) {
+            if (currentPage + showXPages / 2 > lastPage) {
+                end = lastPage;
+                start = Math.max(lastPage - showXPages, 0);
+            } else {
+                start = Math.max(Math.floor(currentPage - showXPages / 2), 0);
+                end = Math.min(start + showXPages, lastPage);
+            }
+
+            _.each(_.range(start, end + 1), (nbr: number): void => {
+                pageSelects.push(
```

Comment:
I didn't know you could push item in a `const` ! I would have used a `let` declaration. nice ! :+1:

Reply:
its because when you push in a array, you keep the same reference so for tslint, you dont reassign the value.